src/map.cpp

```diff
@@ -5083,15 +5083,11 @@ std::list<item> map::use_charges(const tripoint &origin, const int range,
         // if vehicle has FAUCET can use contents from any of the tanks
         if( v->has_part( "FAUCET" ) ) {
             for( auto &pt : v->parts ) {
-                if( pt.is_tank() ) {
-                    for( const auto &obj : pt.contents() ) {
-                        if( obj.typeId() == type ) {
-                            ret.push_back( pt.drain( quantity ) );
-                            quantity -= ret.back().charges;
-                            if( quantity == 0 ) {
-                                return ret;
-                            }
-                        }
+                if( pt.is_tank() && pt.ammo_current() == type ) {
+                    ret.push_back( pt.drain( quantity ) );
```

Comment:
Huh. This was originally checked by the first if-clause in the removed loop. :/

Ironically, this wouldn't happen here now, since the previous check of `ammo_type()` masks the `vehicle_part::drain()` conditional that would return a null item. But you are correct, and that conditional could change.

There aren't (m)any examples of `vehicle_part::drain()` use (it's been added recently, in b899bf05), and I'm very unfamiliar with the code base, so will change [RDY] back to [CR].